discovery-infra/test_infra/utils.py

```diff
@@ -938,3 +938,10 @@ def resource_param(base_value: int, resource_name: str, operators: List):
         return sum((consts.OperatorResource.values()[operator][resource_name] for operator in operators), base_value)
     except KeyError as e:
         raise ValueError(f"Unknown operator name {e.args[0]}")
+
+def disk_names(base: int, count: int):
```

Comment:
Is that really needed? You already do a for loop in `main.tf` where you define the full name of the disks. You can create the prefix over there.

Reply:
It took me some time to figure out how to do it. I use `setproduct` function to find all of the possible combinations of disk names based on worker # and `disk_name` with #.  "You can't dynamically generated nested blocks using for expressions" so we would need to use local variable and reuse it to build `disk_names` but it is not semantically different to what we have now.

Reply:
On the other hand by using range and disk_number I could improve existing expression. Will explore.